internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1455,3 +1459,76 @@ func Install(seq runtime.Sequence, data interface{}) runtime.TaskExecutionFunc {
 		return nil
 	}
 }
+
+// Recover attempts to recover the control plane.
+//
+// nolint: gocyclo
+func Recover(seq runtime.Sequence, data interface{}) runtime.TaskExecutionFunc {
+	return func(ctx context.Context, logger *log.Logger, r runtime.Runtime) (err error) {
+		var (
+			in *machine.RecoverRequest
+			ok bool
+		)
+
+		if in, ok = data.(*machine.RecoverRequest); !ok {
+			return runtime.ErrInvalidSequenceData
+		}
+
+		kubeconfigPath := "/etc/kubernetes/recovery.yaml"
+
+		var b bytes.Buffer
+
+		if err = kubeconfig.GenerateAdmin(r.Config().Cluster(), &b); err != nil {
+			return err
+		}
+
+		if err = ioutil.WriteFile(kubeconfigPath, b.Bytes(), 0600); err != nil {
+			return fmt.Errorf("failed to create recovery kubeconfig: %w", err)
+		}
+
+		// nolint: errcheck
+		defer os.Remove(kubeconfigPath)
+
+		var backend recovery.Backend
+
+		switch in.Source {
+		case machine.RecoverRequest_ETCD:
+			var client *clientv3.Client
+
+			client, err = etcd.NewClient([]string{"127.0.0.1:2379"})
+			if err != nil {
+				return err
+			}
+
+			backend = recovery.NewEtcdBackend(client, "/registry")
+
+		case machine.RecoverRequest_APISERVER:
+			backend, err = recovery.NewAPIServerBackend(kubeconfigPath)
+			if err != nil {
+				return err
+			}
+		}
+
+		as, err := recovery.Recover(context.Background(), backend, kubeconfigPath)
+		if err != nil {
+			return err
+		}
+
+		if err = os.MkdirAll(constants.AssetsDirectory, 0600); err != nil {
+			return err
+		}
+
+		if err = as.WriteFiles(constants.AssetsDirectory); err != nil {
+			return fmt.Errorf("failed to write recovered assets: %w", err)
+		}
+
+		svc := &services.Bootkube{Recover: true}
+
+		system.Services(r).ReloadAndStart(svc)
```

Comment:
This will effectively limit recovery API to the init node since that is the only node that runs bootkube. This needs to be fixed.